Update Apple client with changes from demoable build

jamilbk opened the pull request:
Brings in the changes from the Demoable build so I can start getting feedback from users on.

jamilbk committed:
Add updates from demoable build

jamilbk committed:
Add --deep flag to CodeSigning tool in order to codesign components

jamilbk committed:
Fix ConnlibMock typo

jamilbk committed:
Add multipath entitlement to iOS

francesca64 commented on swift/apple/FirezoneNetworkExtension/PacketTunnelProvider.swift:
```diff
@@ -14,9 +16,13 @@ enum PacketTunnelProviderError: String, Error {
 
-  private lazy var adapter = Adapter(with: self)
+  // Till connlib is updated to expose the desired FFI, we'll use a mock
+  private lazy var adapter = ConnlibMock.Adapter(with: self)
```
Couldn't this be made configurable or something? It's important to be able to test connlib with the Apple client; if a specific pending FFI change is blocking this I can address that ASAP.

jamilbk replied:
That's a good question -- probably better suited for @roop. This was just copy-pasta'd from the `demoable_macos_app` branch in the firezone-apple repo.

I don't think this PR needs merging urgently; I'll mark it as draft and defer to @roop. This was mainly to get the updated UI into TestFlight, but that process is decoupled from our CI pipeline at the moment.

jamilbk replied:
I've reverted everything except the UI changes and it still builds fine, so I'll leave it like this and see if @roop wants to make changes.

roop replied:
I've added commits to make the status bar menu (macOS) update when connlib updates its resources .

jamilbk committed:
Update entitlements and project settings for distribution in TestFlight

jamilbk committed:
Restore Adapter.swift and Bridging Header

jamilbk committed:
Restore PacketTunnelProvider and project file

jamilbk committed:
Remove ConnlibMock since we can now mock in Connlib directly

roop committed:
Tunnel: Suppress calls to unimplemented functions in connlib

Because they will cause the tunnel process to terminate (because
the rust todo()! panics).

roop committed:
Tunnel: Improve how the the CallbackHandler is initialized and stored

roop committed:
Tunnel: Add NetworkResource struct

roop committed:
Tunnel: Bring Adapter in sync with CallbackHandler

roop committed:
Tunnel: Keep track of resources

roop committed:
Tunnel: Reply to app message asking for resources

roop committed:
Make it compile on iOS

roop committed:
Merge branch 'main' into jamilbk-chore/update-apple-with-demo-changes

jamilbk committed:
Add ring to menubar icons; use brand neutral shades (#1811)

Updates the menubar icon to (1) add a ring for better visibility, (2)
use the system status bar size of 22x22 with the icon itself constrained
to 16x16 and (3) use the neutral brand colors from our brand palette in
Figma. Also updates macOS icon with rounded corners and drop shadow to
follow Apple's guidelines

@roop I'm sure I could be overlooking something -- let me know if these
check out or violate any kind of design patterns I wasn't aware of.

## dark menubar disconnected
<img width="52" alt="Screenshot 2023-07-20 at 9 05 34 PM"
src="https://github.com/firezone/firezone/assets/167144/1e8e19e3-bbef-4469-8032-3d6f9ca61ab0">

## dark menubar connected
<img width="59" alt="Screenshot 2023-07-20 at 8 55 54 PM"
src="https://github.com/firezone/firezone/assets/167144/ac495b6f-4985-4f34-97f0-29d126c4e46c">

## light menubar disconnected
<img width="63" alt="Screenshot 2023-07-20 at 9 00 47 PM"
src="https://github.com/firezone/firezone/assets/167144/3ecc1dae-e5b1-41be-94a7-ac3c1062884b">


## light menubar connected
<img width="57" alt="Screenshot 2023-07-20 at 9 00 22 PM"
src="https://github.com/firezone/firezone/assets/167144/6dede923-bbab-48ae-8201-8a6b10a3ca71">

## macOS icon with rounded borders and dropshadow
<img width="374" alt="Screenshot 2023-07-20 at 11 09 39 PM"
src="https://github.com/firezone/firezone/assets/167144/f1f43941-f3e5-4ff1-926c-ce1363f9200d">
